src/scipp/coords.py

```diff
@@ -346,31 +342,28 @@ def show(self, size=None, simplified=False):
 def _log_plan(rules: List[_Rule], targets: Set[str],
               dim_name_changes: Mapping[str, str], coords: _CoordTable) -> None:
     inputs = set(_rule_output_names(rules, _FetchRule))
-    bi_products = {
+    byproducts = {
         name
         for name in (set(_rule_output_names(rules, _RenameRule))
                      | set(_rule_output_names(rules, _ComputeRule))) - targets
         if coords.total_usages(name) < 0
     }
 
-    inputs_str = f'({", ".join(sorted(inputs))})'
-    outputs_str = f'({", ".join(sorted(targets))})'
-    bi_products_str = (f'\n  Bi-products:\n    {", ".join(sorted(bi_products))}'
-                       if bi_products else '')
-
-    dim_rename_steps = '\n'.join(f'    {t} <- {f}' for f, t in dim_name_changes.items())
-    dim_rename_str = ('\n  Rename dimensions:\n' +
-                      dim_rename_steps if dim_rename_steps else '')
-
-    transform_steps = '\n'.join(f'    {rule}' for rule in rules
-                                if not isinstance(rule, _FetchRule))
-    transform_str = '\n  Steps:\n' + transform_steps
+    message = f'Transforming coords ({", ".join(sorted(inputs))}) ' \
+              f'-> ({", ".join(sorted(targets))})'
+    if byproducts:
+        message += f'\n  Byproducts:\n    {", ".join(sorted(byproducts))}'
+    if dim_name_changes:
+        dim_rename_steps = '\n'.join(f'    {t} <- {f}'
+                                     for f, t in dim_name_changes.items())
+        message += '\n  Rename dimensions:\n' + dim_rename_steps
+    message += '\n  Steps:\n' + '\n'.join(
+        f'    {rule}' for rule in rules if not isinstance(rule, _FetchRule))
```

Comment:
Didn't you just convince me that we should keep the `%` syntax?

Reply:
Yes. But I also convinced myself that there is no point in letting the logger format the final message because we have all those `join` calls that have to be done eagerly anyway.